# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/MediaEngineTabVideoSource.cpp
# Commit: f453f2d11453
# Full Hash: f453f2d1145326042d0f9210e2f28f436fde9b74
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-10-07 16:27:18
# Regressor Bug: 1583463
# File Overlap Count: 1
# Description:
#   Bug 1583463 - Don't allocate and append frames when the tab source's stream has been destroyed. r=jib
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D47684
# ==============================================================================

diff -r 0bb60b58c5f4 -r f453f2d11453 dom/media/webrtc/MediaEngineTabVideoSource.cpp
--- a/dom/media/webrtc/MediaEngineTabVideoSource.cpp	Thu Oct 03 19:50:20 2019 +0000
+++ b/dom/media/webrtc/MediaEngineTabVideoSource.cpp	Thu Oct 03 19:50:31 2019 +0000
@@ -257,6 +257,12 @@
     return;
   }
 
+  if (mTrackMain->IsDestroyed()) {
+    // The track was already destroyed by MediaManager. This can happen because
+    // stopping the draw timer is async.
+    return;
+  }
+
   if (mWindow) {
     if (mScrollWithPage || mViewportWidth == INT32_MAX) {
       mWindow->GetInnerWidth(&mViewportWidth);
